src/shared/clients/ec2MetadataClient.ts

```diff
@@ -0,0 +1,12 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export interface InstanceIdentity {
+    region: string
+}
+
+export interface Ec2MetadataClient {
```

Comment:
why do we need these interfaces?

Reply:
Good question - we don't strictly speaking need a client interface here outside of consistency with our other service clients. The response being typed felt nice but given it's laser focused on our current needs it's a bit misleading too. I'd propose to leave this as-is for now and refactor as part of a larger reevaluation of our service clients (do we need them? are we just adding layers of abstraction that add little value?).

Reply:
> outside of consistency with our other service clients.

Unfortunately the existing clients are over-wrought, and we keep copying the patterns there :)



> propose to leave this as-is for now and refactor as part of a larger reevaluation of our service clients (do we need them? are we just adding layers of abstraction that add little value?).

Ok 👍

Reply:
We can replace all the interfaces with types generated from the implementation and remove the `default` files. Won't really change much but it makes it much more ergonomic. You won't need to change things in two places anymore, and using 'Go to definition' takes you to the implementation rather than the interface.